x-pack/legacy/plugins/lens/public/indexpattern_plugin/datapanel.tsx

```diff
@@ -0,0 +1,270 @@
+    );
+  }
+
+  const allFields = props.state.indexPatterns[props.state.currentIndexPatternId].fields;
+  const filteredFields = allFields
+    .filter(
+      (field: IndexPatternField) =>
+        field.name.toLowerCase().includes(nameFilter.toLowerCase()) &&
+        supportedFieldTypes.includes(field.type)
+    )
+    .slice(0, pageSize);
+
+  const availableFieldTypes = _.uniq(filteredFields.map(({ type }) => type));
+  const availableFilteredTypes = typeFilter.filter(type => availableFieldTypes.includes(type));
+
+  return (
+    <ChildDragDropProvider {...props.dragDropContext}>
+      <EuiFlexGroup gutterSize="s" className="lnsIndexPatternDataPanel" direction="column">
+        <EuiFlexItem grow={null}>
+          <div className="lnsIndexPatternDataPanel__header">
+            {!showIndexPatternSwitcher ? (
+              <>
+                <EuiTitle size="xxs">
+                  <h4 title={props.state.indexPatterns[props.state.currentIndexPatternId].title}>
+                    {props.state.indexPatterns[props.state.currentIndexPatternId].title}{' '}
+                  </h4>
+                </EuiTitle>
+                <EuiButtonEmpty
+                  data-test-subj="indexPattern-switch-link"
+                  className="lnsIndexPatternDataPanel__changeLink"
+                  onClick={() => setShowIndexPatternSwitcher(true)}
+                  size="xs"
+                >
+                  (
+                  <FormattedMessage
+                    id="xpack.lens.indexPatterns.changePatternLabel"
+                    defaultMessage="change"
+                  />
+                  )
+                </EuiButtonEmpty>
+              </>
+            ) : (
+              <EuiComboBox
+                data-test-subj="indexPattern-switcher"
+                options={Object.values(props.state.indexPatterns).map(({ title, id }) => ({
+                  label: title,
+                  value: id,
+                }))}
+                inputRef={el => {
+                  if (el) {
+                    el.focus();
+                  }
+                }}
+                selectedOptions={
+                  props.state.currentIndexPatternId
+                    ? [
+                        {
+                          label: props.state.indexPatterns[props.state.currentIndexPatternId].title,
+                          value: props.state.indexPatterns[props.state.currentIndexPatternId].id,
+                        },
+                      ]
+                    : undefined
+                }
+                singleSelection={{ asPlainText: true }}
+                isClearable={false}
+                onBlur={() => {
+                  setShowIndexPatternSwitcher(false);
+                }}
+                onChange={choices => {
+                  setShowIndexPatternSwitcher(false);
+                  props.setState({
+                    ...props.state,
+                    currentIndexPatternId: choices[0].value as string,
```

Comment:
I agree with you on the overuse of useStates. As for clearing, I think it's debatable whether clearing is the right thing. If I want to quickly compare the name of the date fields that start with 'purchasetime' between several indices, it would be pretty handy to keep my filters.

Reply:
I think this is an 80/20% case- which is more convenient in the 80% case? I think your example is a 20% case and implies that every time a user switches indexes they will have to manually deselect all filter types and also clear any text they've typed.